app/controllers/eoi_controller.rb

```diff
@@ -18,4 +19,123 @@ def other_ways_to_help
   def check_feature_flag
     redirect_to "/404" and return unless ENV["FEATURE_EOI_JOURNEY_ENABLED"] == "true"
   end
+
+  def display
+    @application = ExpressionOfInterest.new(session[:eoi])
+
+    # If this new journey is being used, the old journeys (individual and organisation)
+    # aren't applicable any more so delete their cookie data to free up some space in the cookie.
+    if session[:individual_expression_of_interest]
+      session.delete(:individual_expression_of_interest)
+    end
+
+    if session[:organisation_expression_of_interest]
+      session.delete(:organisation_expression_of_interest)
+    end
+
+    step = params["stage"].to_i
+
+    if (1..MAX_STEPS).cover?(step)
+      render path_for_step
+    else
+      redirect_to "/expression-of-interest/self-assessment/start"
+    end
+  end
+
+  def handle_step
+    current_stage = params["stage"].to_i
+    # Pull session data out of session and
+    # instantiate new Application ActiveRecord object
+    @application = ExpressionOfInterest.new(session[:eoi])
+    @application.started_at = Time.zone.now.utc if current_stage == 1
+    # Update Application object with new attributes
+    @application.assign_attributes(application_params)
+
+    if @application.valid?
+      # Update the session
+      session[:eoi] = @application.as_json
+
+      # Replace with routing engine to get next stage
+      next_stage = RoutingEngine.get_next_eoi_step(@application, current_stage)
+
+      if next_stage > MAX_STEPS
+        redirect_to "/expression-of-interest/check_answers"
+      else
+        redirect_to "/expression-of-interest/steps/#{next_stage}"
+      end
+    else
+      render path_for_step
+    end
+  end
+
+  def check_answers
+    @application = ExpressionOfInterest.new(session[:eoi])
```

Comment:
@jrlord does this work even if we don't explicitly call `render eoi/check_answers` ? is it some funky ruby magic and it also knows it's all scoped to the `eoi` views folder?

Reply:
It's just creating a new ExpressionOfInterest instance, passing the form values from the session into the constructor to populate the instance vars.

Reply:
Oh I see what you mean. There's no 'render' call, because it's using implicit rendering, good description here: https://schwad.github.io/ruby/rails/controllers/2017/09/05/take-care-when-using-render-in-rails.html

You've also caused me to notice that the path in the URL is /check_answers not /check-answers, which breaks the established convention. I'll fix that now.